### PR TITLE
Empty implementation of RestAPI

### PR DESCRIPTION
**Pull request by skaupper**
use an object library for objects used by more than one target

### CODE REVIEW

File: src/JukeBox.h [hunk 1]
```diff
@@ -27,7 +27,7 @@
 /**
  * @brief Type for multiple queues
  */
-using Queues = Queue[2];
+using Queues = Queue;
```
Comment (skaupper) [inline src/JukeBox.h]: temporary workaround so it compiles

**Comment by skaupper** [comment]
some changes in this PR, especially to `NetworkListener.h`, will cause merge conflicts with another PRs (#15), so it wont be merged before it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -109,30 +122,45 @@ enable_testing()
...
 # Examples
 ################################################################################
 # Build examples, unless specified otherwise
-if(NOT BUILD_EXAMPLES)
+if(NOT DEFINED BUILD_EXAMPLES)
+  set(BUILD_EXAMPLES TRUE)
+endif()
+
+if(BUILD_EXAMPLES)
+  message(STATUS "Build examples")
```
Comment (wurmmi) [inline CMakeLists.txt]: can we combine these two if() ?

File: CMakeLists.txt [hunk 3]
```diff
@@ -94,10 +101,16 @@ set(DOC_SOURCES         docs/src/main.md
 # Targets
 ################################################################################
 
+# Compile the sources of the main application into an OBJECT library
+#-------------------------------------------------------------------------------
+set(APP_OBJECTS app_object_library)
+add_library(${APP_OBJECTS} OBJECT ${APP_SOURCES} ${APP_HEADER})
+
```
Comment (wurmmi) [inline CMakeLists.txt]: add the `target_compile_options` here as well

File: CMakeLists.txt (hunk 2 above)
Comment (skaupper) [reply CMakeLists.txt]: please see the updated file

**Comment by wurmmi** [comment]
LGTM :+1:

**Pull request merged**